feat(rss): Skip non-distributable images

Takshil-Kunadia opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/1/26890605006346/project/1210562189110460/task/1210629622176531?focus=true .

Adds a new option to RSS feeds that allows publishers to exclude non-distributable images from feed content. The filtering of content uses the same logic as the RTT plugin.

### How to test the changes in this Pull Request:

1. Create a post and add few images - Make sure to add distributable (Can distribute? checked on media) & non-distributable images.
2. Create Feeds from Admin > RSS Feeds.
3. Check for new option - Remove non-distributable images.
4. Enable the option > Click Update > Open the feed url.
5. Check the Distributable image is appearing and non-distributable image is skipped.
6. Also test this option is available only when RTT plugin is active.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

Takshil-Kunadia committed:
feat(rss): add option to remove non-distributable images from feed

leogermani reviewed (commented):
Can we avoid code duplication and create a `remove_non_distrubutable_images( $content )` method in the `Republication_Tracker_Tool_Content` in the RTT plugin, and use that instead?

Just need to extract the part of the code in `get_republishable_content` that does that

Takshil-Kunadia committed:
feat(rss): delegate removal of non-distributable images to Republication Tracker Tool

Takshil-Kunadia commented:
Hi @leogermani, I have implemented the changes. Extracted the image extraction logic into a separate method. Also refactore class-media to separate concerns as "Media distribution" setting of RTT was interfering with the functionality implemented via this PR.

Please do have a look & review of this :- https://github.com/Automattic/republication-tracker-tool/pull/258

Before reviewing this PR 🙇 .Thanks

Takshil-Kunadia committed:
feat(rss): update settings label
